src/redux/books/booksSlice.js

```diff
@@ -40,6 +69,47 @@ const booksSlice = createSlice({
       );
     },
   },
+  extraReducers: (builder) => {
+    // posting a book
+    builder
+      .addCase(postBook.pending, (state) => {
+        state.isLoading = true;
+      })
+      .addCase(postBook.fulfilled, (state) => {
+        state.isLoading = false;
+      })
+      .addCase(postBook.rejected, (state) => {
+        state.isLoading = false;
+      });
+
+    // get books
+    builder
+      .addCase(getBooks.pending, (state) => {
+        state.isLoading = true;
+      })
+      .addCase(getBooks.fulfilled, (state, action) => {
+        state.isLoading = false;
+        const resObject = action.payload;
+
+        const newBooksArr = [];
+        // eslint-disable-next-line no-restricted-syntax, guard-for-in
+        for (const id in resObject) {
+          const bookObj = resObject[id][0];
+          bookObj.item_id = id;
+          newBooksArr.push(bookObj);
+        }
+
+        state.books = newBooksArr;
+      })
+      .addCase(getBooks.rejected, (state) => {
+        state.isLoading = false;
+      });
+
+    // delete books
+    builder.addCase(deleteBook.fulfilled, (state, action) => {
+      console.log("SUCCESS>", action.payload);
```

Comment:
- [Optional] Please avoid committing `console.log` as it's not considered a good practice.

Reply:
Thank you for the feedback. I will make the requested change.